engineering/Testing/defect-template.md

```diff
@@ -0,0 +1,93 @@
+### Defect Report template
+
+### Summary
+
+A brief summary of the bug, mostly within 60 words or below. Make sure your summary is reflecting on what the problem is and where it is.
+
+**What - Where - When (Action)**
+
+### Description:
+
+A detailed description of the bug.
+
+Use the following fields for the description field:
+
+**Reproduce steps:**
+
+Clearly, mention the steps to reproduce the bug.
+Example:
+
+1. Go to '...'
+2. Click on '....'
+3. Scroll down to '....'
+4. Observed result.
+
+**Expected result:** How the application should behave on the above-mentioned steps.
+
+**Actual result:** What is the actual result of running the above steps i.e. the bug behavior.
+
+**Screenshots**
+
+If applicable, add screenshots to help explain your problem.
+
+### Platform/Environment
+
+You may write here the specifications like the version of the project, operating system, or hardware if applicable.
+
+**Desktop (please complete the following information):**
+
+- OS: [e.g. iOS]
+- Browser [e.g. chrome, safari]
+- Version [e.g. 22]
+
+**Smartphone (please complete the following information):**
+
+- Device: [e.g. iPhone6]
+- OS: [e.g. iOS8.1]
+- Browser [e.g. stock browser, safari]
+- Version [e.g. 22]
+
+### Logs / Stack trace
+
+```
+Insert your log/stack trace here
+```
+
+Besides that, we also need to add more about Priority and Severity:
+
+**Priority:** When should a bug be fixed? Priority is generally set from P1 to P5. P1 as “fix the bug with the highest priority” and P5 as ” Fix when time permits”.
+
+**Severity:** This describes the impact of the bug.
+
+**Types of Severity:**
+
+Blocker: No further testing work can be done.
+Critical: Application crash, Loss of data.
+Major: Major loss of function.
+Minor: Minor loss of function.
+Trivial: Some UI enhancements.
+Enhancement: Request for a new feature or some enhancement in the existing one.
+
+===============================================================================
+
+### Some Bonus Tips To Write A Good Bug Report
+
+Given below are some additional tips on how to write a good Bug report:
+
+#1) Report the problem immediately
+If you find any bugs while testing, then you do not need to wait to write a detailed bug report later.
```

Comment:
I think leave the title only. This one "Report the problem immediately" is meaningful enough